src/lib.rs

```diff
@@ -404,7 +411,6 @@ mod explicit {
                     if stack.last().status != OverrideStatus::Isolate && stack.vec.len() >= 2 {
                         stack.vec.pop();
                     }
-                    result.levels[i] = stack.last().level;
```

Comment:
Bug in the spec; see the above comment about the directional status stack.  From line 577 of the reference implementation:

```java
// Rule X7
case PDF:
    // Not really part of the spec
    resultLevels[i] = stack.lastEmbeddingLevel();
```